src/features/game/expansion/components/DiscordBoat.tsx

```diff
@@ -0,0 +1,220 @@
+  };
+
+  const [state, setState] = useState<
+    "connected" | "noDiscord" | "claim" | "claimed"
+  >(initialState());
+
+  const oauth = () => {
+    redirectOAuth();
+  };
+
+  const acknowledge = () => {
+    setState("claim");
+  };
+
+  const claim = () => {
+    // Fire event
+    gameService.send("bonus.claimed", { name: "discord-signup" });
+
+    gameAnalytics.trackMilestone({ event: "Reward:DiscordSignup:Claimed" });
+
+    onClose();
+  };
+
+  if (state === "claimed") {
+    return (
+      <>
+        <div className="p-2">
+          <p className="text-sm mb-2">Wow, nice hat!</p>
+          <p className="text-sm mb-2">
+            {`Don't forget to pay attention to special events and giveaways on
+            Discord so you don't miss out.`}
+          </p>
+        </div>
+      </>
+    );
+  }
+
+  if (state === "claim") {
+    return (
+      <ClaimReward
+        onClaim={claim}
+        reward={{
+          createdAt: Date.now(),
+          id: "discord-bonus",
+          items: BONUSES["discord-signup"].reward.inventory,
+          wearables: BONUSES["discord-signup"].reward.wearables,
+          sfl: 0,
+        }}
+      />
+    );
+  }
+
+  if (state === "connected") {
+    return (
+      <>
+        <div className="p-2">
+          <Label
+            className="mb-2"
+            type="warning"
+            icon={SUNNYSIDE.decorations.treasure_chest}
+          >
+            Bonus reward
+          </Label>
+          <p className="text-xs mb-2">
+            We hope you are enjoying being a part of our community!
+          </p>
+          <p className="text-xs mb-2">
+            {`Pay attention to special events and giveaways on Discord so you
+            don't miss out.`}
+          </p>
+        </div>
+        <Button onClick={acknowledge}>Claim Gift</Button>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <div className="p-2">
+        <Label
+          className="mb-2"
+          type="warning"
+          icon={SUNNYSIDE.decorations.treasure_chest}
+        >
+          Bonus reward
+        </Label>
+        <p className="text-xs mb-2">
+          Did you know there are over 100,000 players in our vibrant Discord
+          community?
+        </p>
+        <p className="text-xs mb-2">
+          If you are looking for farming tips & tricks, it is the place to be.
+        </p>
+        <p className="text-xs mb-2">
+          The best part...everyone who joins gets a free gift!
+        </p>
+      </div>
+      <Button onClick={oauth}>Connect to Discord</Button>
+    </>
+  );
+};
+
+const _isClaimed = (state: MachineState) =>
+  BONUSES["discord-signup"].isClaimed(state.context.state);
+
+const _hasAccess = (state: MachineState) =>
+  hasFeatureAccess(state.context.state, "DISCORD_BONUS");
+
+export const DiscordBoat: React.FC = () => {
+  const [showModal, setShowModal] = useState(false);
+
+  const { authService } = useContext(Auth.Context);
+  const [authState] = useActor(authService);
```

Comment:
You could also use `useSelector` on the authService for the `usesr`. Not crucial given the auth machine doesn't change too much.